Intersect.Client/Entities/Projectiles/Projectile.cs

```diff
@@ -179,109 +183,205 @@ private void AddProjectileSpawns()
+                    }
+                case Direction.UpRight:
+                    switch (projectionDir)
+                    {
+                        case Direction.Up:
+                            return Direction.UpRight;
+                        case Direction.Down:
+                            return Direction.DownLeft;
+                        case Direction.Left:
+                            return Direction.UpLeft;
+                        case Direction.Right:
+                            return Direction.DownRight;
+                        case Direction.UpLeft:
+                            return Direction.Up;
+                        case Direction.UpRight:
+                            return Direction.Right;
+                        case Direction.DownRight:
+                            return Direction.Down;
+                        case Direction.DownLeft:
+                            return Direction.Left;
+                        default:
+                            return projectionDir;
+                    }
+                case Direction.DownLeft:
+                    switch (projectionDir)
+                    {
+                        case Direction.Up:
+                            return Direction.DownLeft;
+                        case Direction.Down:
+                            return Direction.UpRight;
+                        case Direction.Left:
+                            return Direction.DownRight;
+                        case Direction.Right:
+                            return Direction.UpLeft;
+                        case Direction.UpLeft:
+                            return Direction.Down;
+                        case Direction.UpRight:
+                            return Direction.Left;
+                        case Direction.DownRight:
+                            return Direction.Up;
+                        case Direction.DownLeft:
+                            return Direction.Right;
+                        default:
+                            return projectionDir;
+                    }
+                case Direction.DownRight:
+                    switch (projectionDir)
+                    {
+                        case Direction.Up:
+                            return Direction.DownRight;
+                        case Direction.Down:
+                            return Direction.UpLeft;
+                        case Direction.Left:
+                            return Direction.UpRight;
+                        case Direction.Right:
+                            return Direction.DownLeft;
+                        case Direction.UpLeft:
+                            return Direction.Right;
+                        case Direction.UpRight:
+                            return Direction.Down;
+                        case Direction.DownRight:
+                            return Direction.Left;
+                        case Direction.DownLeft:
+                            return Direction.Up;
```

Comment:
This is horrible.

Since I am pretty sure these go in a circle, can you make a static readonly array that caches the "circle" and you just use the two directions to index into that array using a modulo to get the "next point" on the circle?

Reply:
self reminder: double check this tomorrow. ; tried something with no sucess 😢

Reply:
https://github.com/AscensionGameDev/Intersect-Engine/commit/fddf6be4dc2cca42dd1afefaa8e9fd1ca3b87a7b